### PR TITLE
Group Sync Changes to use Settings Attribute vs Env Var.

### PR DESCRIPTION
**Pull request by sabinayakc**
# Fixes the following issue with the release v2.3.12
image

# What's Changed

- Changed the `group_sync` code to use the Settings Attribute vs Environment Variable.
  - `NAUTOBOT_SSO_SUPERUSER_GROUPS --> SSO_SUPERUSER_GROUPS`
  - `NAUTOBOT_SSO_STAFF_GROUPS --> SSO_STAFF_GROUPS`
- Updated `settings.py` to include the full path to the function. from `nautobot.extras.group_sync` to `nautobot.extras.group_sync.group_sync`
# Screenshots
Tested this successfully in local environment. 

image

# TODO

- [X] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [X] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features) #6623 
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Changes look good. Would you mind adding a `changes/6624.fixed` changelog fragment file?

File: changes/6624.fixed [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fixed issue with group_sync.py where it was accessing the settings using environment variable name vs the actual settings name. 
+Also, fixed the SOCIAL_AUTH_PIPELINE settings to include the entire path of the group_sync module.
```
Comment (glennmatthews) [inline changes/6624.fixed:2]:
```suggestion
Fixed issue with `group_sync.py` where it was accessing the settings using environment variable name vs the actual settings name. 
Fixed the `SOCIAL_AUTH_PIPELINE` settings to include the entire path of the `group_sync` function. 
```

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
@cardoe Mind giving this PR a quick look?

**Review by cardoe — APPROVED** [review]
@sabinayakc thank you for the quick follow up here. Really sorry this shipped broken. That's what happens when you make last minute changes that you don't think would break something.

**Pull request merged**